File: Python/選修教學/k731.py
n = int(input())
ox, oy= map(int, input().split())
ori_dir = 0
left, right, back = 0, 0, 0

for i in range(n-1):
    x, y = map(int, input().split())
    
    if x > ox:
        dir = 0  #東
    elif y < oy:
        dir = 1  # 南
    elif x < ox:
        dir = 2  # 西
    else:
        dir = 3  # 北 (y > py)

    if dir == (ori_dir + 1) % 4:
        right += 1
    elif dir == (ori_dir + 2) % 4:
        back += 1
    elif dir == (ori_dir + 3) % 4:
        left += 1

    ori_dir = dir
    ox, oy = x, y

print(left, right, back)

File: APCS/2023/k731.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, x, y, px, py, dir, ori_dir = 0; // original direction, first dir = east
    int left = 0, right = 0, back = 0;
    cin >> n >> px >> py;

    for (int i = 1; i < n; i++) {
        cin >> x >> y;

        if (x > px) dir = 0; // East
        else if (y < py) dir = 1; // South
        else if (x < px) dir = 2; // West
        else dir = 3; // North 

        if (dir == (ori_dir + 1) % 4) right++;
        else if (dir == (ori_dir + 2) % 4) back++;
        else if (dir == (ori_dir + 3) % 4) left++;

        ori_dir = dir;

        px = x, py = y;
    }

    cout << left << ' ' << right << ' ' << back << '\n';
    return 0;
}
